File: 백준/Silver/10828. 스택/스택.py
import sys
from collections import deque

input = sys.stdin.readline

N = int(input())
stack = deque()
result = list()

for _ in range(N):
    line = list(input().strip().split())
    if line[0] == "push":
        stack.append(int(line[1]))
    elif line[0] == "pop":
        if stack:
            result.append(stack.pop())
        else:
            result.append(-1)
    elif line[0] == "size":
        result.append(len(stack))
    elif line[0] == "empty":
        if stack:
            result.append(0)
        else:
            result.append(1)
    elif line[0] == "top":
        if stack:
            result.append(stack[-1])
        else:
            result.append(-1)

sys.stdout.write('\n'.join(map(str, result)) + '\n')

File: 백준/Silver/10828. 스택/스택.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        int N = Integer.parseInt(br.readLine().trim());

        Stack<Integer> stack = new Stack<>();
        StringBuilder sb = new StringBuilder();

        for(int i = 0; i < N; i++){
            String[] line = br.readLine().split(" ");
            if(line[0].equals("push")){
                stack.push(Integer.parseInt(line[1]));
            }else if(line[0].equals("pop")){
                if (stack.empty()) sb.append(-1).append('\n');
                else sb.append(stack.pop()).append('\n');
            }else if(line[0].equals("size")){
                sb.append(stack.size()).append('\n');
            }else if(line[0].equals("empty")){
                if (stack.empty()) sb.append(1).append('\n');
                else sb.append(0).append('\n');
            }else if(line[0].equals("top")){
                if (stack.empty()) sb.append(-1).append('\n');
                else sb.append(stack.peek()).append('\n');
            }
        }
        
        bw.write(sb.toString());
        bw.flush();
        bw.close();
        br.close();
    }
}